libs/blocks/global-navigation/global-navigation.js

```diff
@@ -251,6 +259,45 @@ const convertToPascalCase = (str) => str
   .map((s) => s.charAt(0).toUpperCase() + s.slice(1))
   .join(' ');
 
+// TODO: debate whether this should be here or in a separate file,
+// since it's not time-sensitive and we want to potentially bundle it
+// TODO: position the prompt relative to the App Switcher
+const loadAppPrompt = async () => {
```

Comment:
Should this be available on all pages? I was thinking it will be something that gets loaded via metadata and then the logic is executed. I am all for moving this to a separate file since it will only apply to a few pages.

Reply:
Yes, agree with Narcis. I also wonder whether this should be in the `webapp-prompt.js` directly and you pass a few more (potentially optional) properties?
It seems like base-setup for this to work, so any consumer would also need to retrieve this data, correct?

Reply:
The thing I was trying to avoid is loading a module on a lot of pages, when the feature would actually be used on a very small subset of pages. The checks here are 35 lines long and ensure that we're loading `webapp-prompt.js` only if actually required. Moving them to the module will import some 200 lines, although they might not be used.

As a compromise, I'm thinking of keeping the metadata/signed-in/device checks here and moving the dismissal state and entitlement checks to the `webapp-prompt.js` file. Do you think that's a good compromise?

Reply:
Definitely a tradeoff, yeah lets go with your suggestion for now @overmyheadandbody

Reply:
I've split the logic between `global-navigation.js` and `webapp-prompt.js` so that the footprint in the former is as small as possible, while still avoiding to load the PEP logic if not really needed.